app/src/androidTest/java/org/andstatus/todoagenda/BirthdayTest.java

```diff
@@ -19,31 +18,34 @@
 
     private static final String TAG = BirthdayTest.class.getSimpleName();
 
-    private MockCalendarContentProvider provider = null;
+    private MockWidget widget = null;
+    private MockContentProviderImpl provider = null;
```

Comment:
I think splitting MockCalendarContentProvider into two classes is not necessary, it only increases number of lines to use it and causes lots of unneeded changes in tests thus making it much harder to see the essence of the proposed change :-(
Moreover, "MockWidget" and widget seem to me less fitting to their real meaning...

Reply:
Here I have to disagree with you. MockCalendarContentProvider did much more than being a mock content provider for calendar - it was a whole test harness that creates a Context, simulates some necessary features of the widget (loading state, editing settings, fetching and building the event list, etc).

And on a practical note, I needed a new mocked ContentProvider for tasks - there's even a test that uses calendar and tasks at the same time, and splitting only the ContentProvider part allowed that class to be reused, and the test harness part to be used in the new task tests.

For what's worth, the changes to existing tests were basically just the refactorings - as far as I remember it was not necessary to make any significant changes to them.

Reply:
The problem is that I don't like these refactoring :-(
As current app maintainer I assure you that having your PR with addition of the new feature without any refactoring will help future app maintainance greatly.
Just extend existing code, I will refactor everything later. Please.
And please squash all your changes into one commit.